### PR TITLE
Converge internal and external ui-extensions api

### PR DESCRIPTION
**Pull request by zeusdeux**
- [x] Expose `contentType`
- [x] Add private `setString` method
- [x] Change `setValue` to call `setString` or `setValue` based on type of widget
- [x] Update `signal` implementation to be in line with what's in UI repo
- [x] Expose `space.getPublishedEntries`
- [x] Expose `space.getPublishedAssets`
- [x] Fix broken `api.field.onValueChanged` behaviour ~~(blocked till integration tests work so that we can test this breaking change properly)~~
- [x] Expose `field.removeValue`
- [x] Expose `field.onIsDisabledChanged`
- [x] Expose `field.setActive`
- [x] Expose `field.setInvalid`
- [x] Update changelog
- [x] Update docs

### CODE REVIEW

**Comment by zeusdeux** [comment]
CI failing since I haven't pulled in `master` with phantom-js related changes yet. Will do so after review.

File: CHANGELOG [hunk 1]
```diff
@@ -3,14 +3,32 @@ All notable changes to this project will be documented in this file.
...
+    - Expose `api.field.type`
+    - Expose `api.entry.field[id].onIsDisabledChanged(locale, handler)`
+
+### Changed
+
+    - Field focus/blur status is communicated internally to the Contentful web application for the styles to work correctly.
+
+### Fixed
+
+    - Fix `api.field.onValueChanged` firing on the tab that _performed_ the update. This was not in line with documented behaviour.
```
Comment (geigerzaehler) [inline CHANGELOG]: Be more precise. `onValueChanged` fired after calling `set()`.

File: doc/widget-api-frontend.md [hunk 2]
```diff
@@ -58,6 +59,46 @@ contentfulWidget.init(function (widget) {
...
+  "id": "<id of field>",
+  "name": "<name of field>",
+  "type": "<type of field. e.g., Symbol, Text, etc>",
+  "localized": <boolean>,
+  "required": <boolean>,
+  "validations": [<validations>],
+  "disabled": <boolean>,
+  "omitted": <boolean>
+}
+~~~
```
Comment (geigerzaehler) [inline doc/widget-api-frontend.md]: Instead we could just link to the API documentation.

File: doc/widget-api-frontend.md [hunk 3]
```diff
@@ -92,13 +133,25 @@ has been acknowledged. The type of the value must match the expected field type.
...
+
 ##### `widget.field.onValueChanged(cb): function`
 Calls the callback every time the value of the field is changed by some external
 event (e.g. when multiple editors are working on the same entry). It will not be
 called after `setValue()` is called.
 
 The method returns a function that can be called to stop listening to changes.
 
+##### `widget.field.onIsDisabledChanged(cb): function`
+Calls the callback when the disabled status of the field changes.
```
Comment (geigerzaehler) [inline doc/widget-api-frontend.md]: Please indicate that a boolean is passed to the callback.

File: doc/widget-api-frontend.md [hunk 4]
```diff
@@ -107,6 +160,8 @@ example.
 The current locale of a field the widget is attached to. Yields `"en_US"` in the
 example.
 
+##### `widget.field.type: string`
+Gives the field type of the widget.
```
Comment (geigerzaehler) [inline doc/widget-api-frontend.md]: “Holds the type of the field the widget is attached to”. Please also link to the CMA documentation for a list of all field types.

**Comment by geigerzaehler** [comment]
Please update the ref docs to metnion and point to version 0.8 of `contentful-management.js`.

File: CHANGELOG [hunk 5]
```diff
@@ -3,14 +3,32 @@ All notable changes to this project will be documented in this file.
 This project adheres to [Semantic Versioning](http://semver.org/) and
 the [“Keep a Changelog” format](http://keepachangelog.com/).
 
+## Unreleased - 2015-06-15
+
+### Added
+
+    - Expose `api.contentType` that contains name of content type used, data on all the fields in the entry, current display field and sys data.
```
Comment (geigerzaehler) [inline CHANGELOG]: We use `widget` instead of `api` in the docs.

**Comment by geigerzaehler** [comment]
Could you also rename `CHANGELOG` to `CHANGELOG.md`?

File: doc/widget-api-frontend.md (hunk 4 above)
Comment (zeusdeux) [inline doc/widget-api-frontend.md]: Are we calling 'em widgets?

File: doc/widget-api-frontend.md (hunk 3 above)
Comment (zeusdeux) [inline doc/widget-api-frontend.md]: ✅

File: doc/widget-api-frontend.md [hunk 6]
```diff
@@ -62,42 +62,7 @@ contentfulWidget.init(function (widget) {
...
-  "name": "<name of field>",
-  "type": "<type of field. e.g., Symbol, Text, etc>",
-  "localized": <boolean>,
-  "required": <boolean>,
-  "validations": [<validations>],
-  "disabled": <boolean>,
-  "omitted": <boolean>
-}
-~~~
+It has the shape as described under "content type properties" [here](https://www.contentful.com/developers/docs/references/content-management-api/#/reference/content-types).
```
Comment (geigerzaehler) [inline doc/widget-api-frontend.md]: Please don’t use “here” link as [this Smashing Mag post](https://www.smashingmagazine.com/2012/06/links-should-never-say-click-here/) explains. Also applies to `widget.field.type` section

File: doc/widget-api-frontend.md [hunk 7]
```diff
@@ -170,7 +137,7 @@ entry and to get the entry's metadata.
 
 ##### `entry.getSys(): object`
 Returns metadata for an entry. The value coincides with the `sys` value of an
-entry returned by the [Contentful Management API](https://github.com/contentful/contentful-management.js#entry-properties)
+entry returned by the [Contentful Management API](https://github.com/contentful/contentful-management.js/tree/legacy)
```
Comment (geigerzaehler) [inline doc/widget-api-frontend.md]: Keep the URL fragment. This way the user gets to the right section immediately

File: CHANGELOG.md [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+    - Expose `widget.field.type`
+    - Expose `widget.entry.field[id].onIsDisabledChanged(locale, handler)`
+
+### Changed
+
+    - Field focus/blur status is communicated internally to the Contentful web application for the styles to work correctly.
+
+### Fixed
+
+    - Fix `widget.field.onValueChanged` firing on the tab that called `field.setValue()`.
```
Comment (geigerzaehler) [inline CHANGELOG.md]: No clue what “tab” means. Please explain the problem more thoroughly

**Comment by geigerzaehler** [comment]
> Please update the ref docs to metnion and point to version 0.8 of contentful-management.js.

Could you make the version explicit in the docs? Every where we mention the library we should say that it is v0.8 so readers don’t get confused when they are already familiar with the new version.

**Comment by zeusdeux** [comment]
Squashing.

**Pull request merged**